include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,555 @@
+{
+private:
+    //!\brief The iterator type of the underlying range.
+    using it_t = std::ranges::iterator_t<rng_t>;
+    //!\brief The sentinel type of the underlying range.
+    using sentinel_t = std::ranges::sentinel_t<rng_t>;
+
+public:
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief Type for distances between iterators.
+    using difference_type = typename std::iter_difference_t<it_t>;
+    //!\brief Value type of this iterator.
+    using value_type = size_t;
+    //!\brief The pointer type.
+    using pointer = void;
+    //!\brief Reference to `value_type`.
+    using reference = value_type;
+    //!\brief Tag this class as input iterator.
+    using iterator_category = std::input_iterator_tag;
+    //!\brief Tag this class depending on which concept `it_t` models.
+    using iterator_concept = std::conditional_t<std::contiguous_iterator<it_t>,
+                                                typename std::random_access_iterator_tag,
+                                                seqan3::iterator_tag_t<it_t>>;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr window_iterator()                                    = default; //!< Defaulted.
+    constexpr window_iterator(window_iterator const &)             = default; //!< Defaulted.
+    constexpr window_iterator(window_iterator &&)                  = default; //!< Defaulted.
+    constexpr window_iterator & operator=(window_iterator const &) = default; //!< Defaulted.
+    constexpr window_iterator & operator=(window_iterator &&)      = default; //!< Defaulted.
+    ~window_iterator()                                             = default; //!< Defaulted.
+
+    /*!\brief Construct from begin and end iterators of a given range over std::integral elements, and the number of
+    *                      elements per window.
+    * /param[in] it_start Iterator pointing to the first position of the std::integral range.
+    * /param[in] it_end   Iterator pointing to the last position of the std::integral range.
+    * /param[in] w        The number of elements in one window.
+    *
+    * \details
+    *
+    */
+    window_iterator(it_t it_start, sentinel_t it_end, uint32_t w) :
+                    last_elem{it_end}, window_left{it_start}, window_right{it_start},
+                    num_w_elems{w}
+    {
+        if (num_w_elems > std::ranges::distance(window_left, last_elem))
+            num_w_elems = std::ranges::distance(window_left, last_elem);
+        if (window_right != last_elem)
+            get_minimiser();
+    }
+    //!\}
+
+    //!\anchor window_iterator_comparison
+    //!\name Comparison operators
+    //!\{
+
+    //!\brief Compare to iterator on underlying range.
+    friend bool operator==(window_iterator const & lhs, sentinel_t const & rhs) noexcept
+    {
+        return lhs.window_right == rhs;
+    }
+
+    //!\brief Compare to iterator on underlying range.
```

Comment:
You are not comparing to an iterator of the underlying range here?

Reply:
Is that not what the sentinel is? In kmer_hash this is commented as "Compare to iterator on text."

Reply:
Ah I see the confusion. The end of the minimizer range is also the end of the underlying range, thats why the sentinel type is the same. This comment suggests that I can compare to any iterator of the underlying range, but I can only compare to the sentinel.
How about `Compare to the sentinel of the underlying range.`